nmdc_runtime/site/resources.py

```diff
@@ -194,15 +194,17 @@ def get_object_bytes(self, object_id) -> requests.Response:
             access = AccessURL(
                 **self.get_object_access(object_id, method.access_id).json()
             )
-            if access.url.startswith(
+            if str(access.url).startswith(
                 os.getenv("API_HOST_EXTERNAL")
             ) and self.base_url == os.getenv("API_HOST"):
-                access.url = access.url.replace(
-                    os.getenv("API_HOST_EXTERNAL"), os.getenv("API_HOST")
+                access.url = AnyUrl(
+                    str(access.url).replace(
+                        os.getenv("API_HOST_EXTERNAL"), os.getenv("API_HOST")
+                    )
                 )
         else:
             access = AccessURL(url=method.access_url.url)
-        return requests.get(access.url)
+        return requests.get(str(access.url))
```

Comment:
Side note: Instead of casting it 3x, we could store the casted result as `access_url_str`.

Reply:
yeah, I considered that. we get some url (format) validation by keeping to the model until the return.